src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/PSScoreUtils.java

```diff
@@ -0,0 +1,426 @@
+package org.broadinstitute.hellbender.tools.spark.pathseq;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import htsjdk.samtools.*;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.broadcast.Broadcast;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.gcs.BucketUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import scala.Tuple2;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.Charset;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public final class PSScoreUtils {
+
+    public static final String HITS_TAG = "YP";
+    public static final double SCORE_NORMALIZATION_FACTOR = 1e6;
+    private static final Logger logger = LogManager.getLogger(PSScoreUtils.class);
+
+    public static JavaRDD<GATKRead> scoreReads(final JavaSparkContext ctx,
+                                               final JavaRDD<GATKRead> pairedReads,
+                                               final JavaRDD<GATKRead> unpairedReads,
+                                               final SAMFileHeader header,
+                                               final PSScoreArgumentCollection scoreArgs) {
+
+        //Group reads into pairs
+        final JavaRDD<Iterable<GATKRead>> groupedReads = PSScoreUtils.groupReadsIntoPairs(pairedReads,
+                unpairedReads, scoreArgs.readsPerPartition);
+
+        //Load taxonomy database, created by running PathSeqBuildReferenceTaxonomy with this reference
+        final PSTaxonomyDatabase taxDB = PSScoreUtils.readTaxonomyDatabase(scoreArgs.taxonomyDatabasePath);
+        final Broadcast<Map<String, String>> accessionToTaxIdBroadcast = ctx.broadcast(taxDB.accessionToTaxId);
+
+        //Check header against database
+        if (scoreArgs.headerWarningFile != null) {
```

Comment:
They happen frequently because the NCBI taxonomy database and RefSeq releases are usually out of sync, so you often get a lot of accessions that are in the reference but have since been discovered to actually be a duplicate of some other organism. The problem is that the list of accessions can sometimes number in the thousands (for large samples) so the screen gets flooded.